cmd/platform/sampledata.go

```diff
@@ -0,0 +1,628 @@
+		return errors.New("You can't have more channel memberships than channels per team.")
+	}
+
+	var bulkFile *os.File
+	switch bulk {
+	case "":
+		bulkFile, err = ioutil.TempFile("/tmp", "mattermost-sample-data-")
+		defer os.Remove(bulkFile.Name())
+		if err != nil {
+			return errors.New("Unable to open temporary file.")
+		}
+	case "-":
+		bulkFile = os.Stdout
+	default:
+		bulkFile, err = os.OpenFile(bulk, os.O_RDWR|os.O_CREATE, 0755)
+		if err != nil {
+			return errors.New("Unable to write into the \"" + bulk + "\" file.")
+		}
+	}
+
+	encoder := json.NewEncoder(bulkFile)
+	version := 1
+	encoder.Encode(app.LineImportData{Type: "version", Version: &version})
+
+	fake.Seed(seed)
+	rand.Seed(seed)
+
+	teamsAndChannels := make(map[string][]string)
+	for i := 0; i < teams; i++ {
+		teamLine := createTeam(i)
+		teamsAndChannels[*teamLine.Team.Name] = []string{}
+		encoder.Encode(teamLine)
+	}
+
+	teamsList := []string{}
+	for teamName := range teamsAndChannels {
+		teamsList = append(teamsList, teamName)
+	}
+	sort.Strings(teamsList)
+
+	for _, teamName := range teamsList {
+		for i := 0; i < channelsPerTeam; i++ {
+			channelLine := createChannel(i, teamName)
+			teamsAndChannels[teamName] = append(teamsAndChannels[teamName], *channelLine.Channel.Name)
+			encoder.Encode(channelLine)
+		}
+	}
+
+	allUsers := []string{}
+	for i := 0; i < users; i++ {
+		userLine := createUser(i, teamMemberships, channelMemberships, teamsAndChannels, profileImages)
+		encoder.Encode(userLine)
+		allUsers = append(allUsers, *userLine.User.Username)
+	}
+
+	for team, channels := range teamsAndChannels {
+		for _, channel := range channels {
+			for i := 0; i < postsPerChannel; i++ {
+				postLine := createPost(team, channel, allUsers)
+				encoder.Encode(postLine)
+			}
+		}
+	}
+
+	for i := 0; i < directChannels; i++ {
+		user1 := allUsers[rand.Intn(len(allUsers))]
+		user2 := allUsers[rand.Intn(len(allUsers))]
+		channelLine := createDirectChannel([]string{user1, user2})
+		encoder.Encode(channelLine)
+		for j := 0; j < postsPerDirectChannel; j++ {
+			postLine := createDirectPost([]string{user1, user2})
+			encoder.Encode(postLine)
+		}
+	}
+
+	for i := 0; i < groupChannels; i++ {
+		users := []string{}
+		totalUsers := 3 + rand.Intn(3)
+		for len(users) < totalUsers {
+			user := allUsers[rand.Intn(len(allUsers))]
+			if !sliceIncludes(users, user) {
+				users = append(users, user)
+			}
+		}
+		channelLine := createDirectChannel(users)
+		encoder.Encode(channelLine)
+		for j := 0; j < postsPerGroupChannel; j++ {
+			postLine := createDirectPost(users)
+			encoder.Encode(postLine)
+		}
+	}
+
+	if bulk == "" {
+		_, err := bulkFile.Seek(0, 0)
+		if err != nil {
+			return errors.New("Unable to read correctly the temporary file.")
+		}
+		importErr, lineNumber := a.BulkImport(bulkFile, false, workers)
+		if importErr != nil {
+			return errors.New(fmt.Sprintf("%s: %s, %s (line: %d)", importErr.Where, importErr.Message, importErr.DetailedError, lineNumber))
+		}
```

Comment:
Is it intentional that if the `--bulk` flag is specified, we don't import the sample data? If so, that should be probably made clear in the flags documentation above.

Reply:
Explained in the help